client/plugins/userLibrary.ts

```diff
@@ -0,0 +1,87 @@
+import { UserLibraryApiGetLatestMediaRequest } from '@jellyfin/client-axios';
+import { Plugin } from '@nuxt/types/app';
+
+type GetLatestMediaParams = Omit<UserLibraryApiGetLatestMediaRequest, 'userId'>;
+
+type UserLibraryType = {
+  getItem: (id: string, force?: boolean) => Promise<void>;
+  getLatestMedia: (params: GetLatestMediaParams) => Promise<string[]>;
+};
+
+declare module '@nuxt/types' {
+  interface Context {
+    $userLibrary: UserLibraryType;
+  }
+
+  interface NuxtAppOptions {
+    $userLibrary: UserLibraryType;
+  }
+}
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    $userLibrary: UserLibraryType;
+  }
+}
+
+declare module 'vuex/types/index' {
+  // eslint-disable-next-line -- Current TypeScript rules flag S as unused, but Nuxt requires identical types
+  interface Store<S> {
+    $userLibrary: UserLibraryType;
+  }
+}
+
+const userLibraryPlugin: Plugin = ({ $api, $auth, store }, inject) => {
+  inject('userLibrary', {
+    /**
+     * Fetches an item based on its ID and stores it
+     *
+     * @param {string} id - Item ID
+     * @param {boolean} force - Force fetch
+     */
+    getItem: async (id: string, force = true): Promise<void> => {
```

Comment:
```suggestion
    getItem: async (id: string): Promise<void> => {
```

As said previously: don't do these checks in here. This is purely a wrapper around the API. The belong wherever this is called, we don't want to abstract this away, it needs to be clear for new contributors.

How do we do something different if the item already exists?

It would also be the only function in the API to do this, which is super inconsistent. Just leave it to the pages/components to check for this.